src/Tasks/Microsoft.NET.Build.Tasks/ConflictResolution/ReferenceComparer.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.NET.Build.Tasks.ConflictResolution
+{
+    internal class ReferenceComparer<T> : IEqualityComparer, IEqualityComparer<T>
```

Comment:
Probably.  At one point I was putting ITaskItems into a dictionary or hashset and wanted reference equality (rather than whatever the backing ITaskItem may have implemented).  I believe all cases that were doing this before now wrap the item in a ConflictItem so this isn't needed.